samples/vboxwrapper/vbox_vboxmanage.cpp

```diff
@@ -2236,3 +2291,248 @@ bool VBOX_VM::is_hostrtc_set_to_utc() {
+                    vboxlog_msg("Error: ERR_TIMEOUT");
+                    vboxlog_msg("Attempts: %d", attempts);
+
+                    fd_race_mitigator = 0;
+                    return ERR_TIMEOUT;
+                }
+                boinc_sleep(sleep_low + sleep_high * drand());
+                attempts++;
+            } else {
+                vboxlog_msg("Could not set race mitigation lock.");
+                vboxlog_msg("Lockname: '%s'", lock_name.c_str());
+                vboxlog_msg("Error: %d, %s", err, strerror(err));
+                vboxlog_msg("Attempts: %d", attempts);
+
+                fd_race_mitigator = 0;
+                return ERR_FOPEN;
+            }
+        }
+    }
+#endif
+    return BOINC_SUCCESS;
+}
+
+// To remove the parent disk entry use 'closemedium disk "<path to disk>"' instead of 'closemedium disk "<UUID>"'
+// since the latter sometimes returns an error like this:
+//
+// VBoxManage closemedium disk "1d9935fd-37c9-4c34-946b-f6d252c6a1af"
+// VBoxManage: error: The given path '1d9935fd-37c9-4c34-946b-f6d252c6a1af' is not fully qualified
+// VBoxManage: error: Details: code VBOX_E_FILE_ERROR (0x80bb0004), component MediumWrap, interface IMedium, callee nsISupports
+// VBoxManage: error: Context: "OpenMedium(Bstr(pszFilenameOrUuid).raw(), enmDevType, enmAccessMode, fForceNewUuidOnOpen, pMedium.asOutParam())" at line 197 of file VBoxManageDisk.cpp
+//
+// Output of 'VBoxManage showhdinfo "/path to/disk/parent_disk_name.vdi"' usually looks like this:
+//
+// UUID:           f81c0950-57ee-462e-b931-051193700d76
+// Parent UUID:    base
+// State:          created
+// Type:           multiattach
+// Location:       /path to/disk/parent_disk_name.vdi
+// Storage format: VDI
+// Format variant: dynamic default
+// Capacity:       51200 MBytes
+// Size on disk:   2 MBytes
+// Encryption:     disabled
+// Property:       AllocationBlockSize=1048576
+// Child UUIDs:    67e34269-e52e-4957-a813-c85f72084fba
+//                 5fa7905e-72e4-4f85-8405-dc6401417720
+//                 7262b5e5-2b32-482a-8c6e-2ef49548740b
+//
+int VBOX_VM::remove_vbox_disk_orphans(string vbox_disk) {
+    int retval;
+    string command;
+    string output;
+    string childlist;
+    string child_uuid;
+    size_t childlist_start;
+    string loc_line;
+    size_t loc_start;
+    size_t loc_end;
+
+    command  = "showhdinfo \"" + vbox_disk + "\" ";
+    retval = vbm_popen(command, output, "hdd registration", false, false);
+    if (retval) {
+        vboxlog_msg("Could not get disk details in 'remove_vbox_disk_orphans'.");
+        return retval;
+    }
+    output = output + "\n";
+
+    childlist_start = output.find("\nChild UUIDs:");
+    if (childlist_start != string::npos) {
+        size_t pos = 0;
+
+        childlist = output.substr(childlist_start + 17, string::npos - childlist_start);
+
+        while (pos < childlist.size()) {
+            child_uuid = childlist.substr(pos, 36);
```

Comment:
Is it possible to get rid of this magic number or at least explain what does it mean?

Reply:
done